imap_processing/codice/make_xtce.py

```diff
@@ -0,0 +1,237 @@
+""" The module will take an Excel file and convert it into an XTCE formatted XML file.
+
+"""
+import xml.etree.ElementTree as ET
+
+import pandas as pd
+
+sheet_name = "P_COD_AUT"  # Assuming the sheet name is correct
+
+# Create CCSDS Header parameters. This should be consistent for all CCSDS packets.
+ccsds_parameters = [
+    {
+        "name": "VERSION",
+        "parameterTypeRef": "uint3",
+        "description": "CCSDS Packet Version Number (always 0)",
+    },
+    {
+        "name": "TYPE",
+        "parameterTypeRef": "uint1",
+        "description": "CCSDS Packet Type Indicator (0=telemetry)",
+    },
+    {
+        "name": "SEC_HDR_FLG",
+        "parameterTypeRef": "uint1",
+        "description": "CCSDS Packet Secondary Header Flag (always 1)",
+    },
+    {
+        "name": "PKT_APID",
+        "parameterTypeRef": "uint11",
+        "description": "CCSDS Packet Application Process ID",
+    },
+    {
+        "name": "SEG_FLGS",
+        "parameterTypeRef": "uint2",
+        "description": "CCSDS Packet Grouping Flags (3=not part of group)",
+    },
+    {
+        "name": "SRC_SEQ_CTR",
+        "parameterTypeRef": "uint14",
+        "description": "CCSDS Packet Sequence Count (increments with each new packet)",
+    },
+    {
+        "name": "PKT_LEN",
+        "parameterTypeRef": "uint16",
+        "description": "CCSDS Packet Length (number of bytes after Packet length minus 1)",
+    },
+    {
+        "name": "SHCOARSE",
+        "parameterTypeRef": "uint32",
+        "description": "CCSDS Packet Time Stamp (coarse time)",
+    },
+]
+
+if __name__ == "__main__":
+    # ET.register_namespace is important! Make sure you use the correct namespace for the xtce file you are using.This
+    # is the namespace for the IMAP xtce files currently.
+
+    ET.register_namespace("xtce", "http://www.omg.org/space")
+
+    # Load data from Excel file
+    # This is important! Use the xls PATH to the file you want to convert to XML.
+    # Also, make sure the "sheet" name is correct. Case sensitive.'''
+
+    xls = pd.ExcelFile(
+        "/Users/gamo6782/Desktop/IMAP/TLM_COD_20230629-110638(update).xlsx"
+    )
+    df = xls.parse(sheet_name)
+
+    # Fill missing values with '-*-'
+    df.fillna("", inplace=True)
+
+    # Create the root element and add namespaces
+    root = ET.Element(
+        "{http://www.omg.org/space}SpaceSystem",
+        nsmap={"xtce": "http://www.omg.org/space"},
+    )
+    root.attrib["name"] = "P_COD_AUT"
+
+    # Create the Header element
+    header = ET.SubElement(root, "{http://www.omg.org/space}Header")
+    header.attrib["date"] = "2023-08-21"
+    header.attrib["version"] = "1.0"
+    header.attrib["author"] = "Gabriel Moraga"
```

Comment:
These are some more hard-coded parameters that could be passed into the script in another way.  I think we should have a conversation with the team on how much we want to 'generalize' this code -- If we expect multiple people on the team to be using this, then we probably should explore ways to avoid hard-coded parameters.

Reply:
I added this above it for now:
" #Create the Header element with attributes 'date', 'version', and 'author' "